src/libs/ajax/loaded-data/useLoadedData.ts

```diff
@@ -31,7 +44,8 @@ export type UseLoadedDataResult<T> = [
  * }
  * @returns a tuple with [currentLoadedState, updateDataMethod]
  */
-export const useLoadedData = <T>(): UseLoadedDataResult<T> => {
+export const useLoadedData = <T>(hookArgs?: UseLoadedDataArgs<T>): UseLoadedDataResult<T> => {
+  const args: UseLoadedDataArgs<T> = hookArgs ? hookArgs : {}
```

Comment:
Is it not possible for the type to be inferred here?
```suggestion
  const arg = hookArgs ? hookArgs : {}
```

Reply:
this is for added type safety so that any future code changes don't change the attended outcome of a usable set of args according to the type provided.  Think of scenarios where a required arg is added to the type later... then the fallback of empty object {} would be a wanted type error here.

Reply:
Could a default parameter value be used here? If so, it seems like that would fix a few problems.
```
export const useLoadedData = <T>(args: UseLoadedDataArgs<T> = {}): UseLoadedDataResult<T> => {
```

Reply:
good call!  Let me try that...

Reply:
done